BLD fix circle build on 0.19.X

jnothman opened the pull request:
In order to release 0.19.2 with Python 3.7 support, we should probably make sure we can recompile docs.

I've put in a couple of fixes, but I've not yet worked out how to fix `fetch_mldata` failures. How are these working in master? Are we relying on a `scikit_learn_data` cache in master?

jnothman committed:
FIX n_iter_ should be less than max_iter in when using lbfgs solver(#10723)

iArunava committed:
EXA Changed deprecated method set_axis_bgcolor to set_facecolor. (#10763)

oleksandr-pavlyk committed:
MAINT: correct cythonization of sklearn.metrics.cluster to avoid warning (#10851)

lesteve committed:
EXA Fix plot tomography l1 reconstruction (#11100)

* Ensure that the number of angles is integer

* Fix plot_tomography_l1_reconstruction.py

Error was: TypeError: 'numpy.float64' object cannot be interpreted as an integer.

[doc build]

Commit:
Compatibility with python 3.7.0b5 #11224 (#11256)

jnothman committed:
Try upgrading python version requirements in CI

jnothman committed:
Pin Cython to 0.27.3

jnothman committed:
Update version to 0.19.2

ogrisel committed:
FIX use C:\Python37 on appveyor

ogrisel committed:
Fix the cython version on travis/ubuntu

ogrisel committed:
DOC better comments

ogrisel committed:
FIX doctest ellipsis pattern

jnothman committed:
Install Cython and Pandas with pip on Travis

jnothman committed:
Let Appveyor use most recent versions

jnothman committed:
Travis needs to stall scipy with pip too

ogrisel committed:
MAINT Install nose and pytest with pip

jnothman committed:
Don't test pandas for now

jnothman committed:
TST Skip doctests for numpy < 1.14

jnothman committed:
Revert circle changes so that docs build on working config

jnothman committed:
Copy plot_stock_market from master

jnothman committed:
cm.spectral -> cm.nipy_spectral

qinhanmin2014 commented:
Some pull requests needed here according to the Circle log:
#11100 for plot_tomography_l1_reconstruction.py
#10763 for plot_sparse_cov.py
#10958 We've decided to add ``- PANDAS_VERSION: 0.13.1`` to Circle config

Honestly I suggest that we move to Circle 2 (#10167) since I see the following things:
This project is currently running on CircleCI 1.0 which will no longer be supported after August 31, 2018. Please start migrating this project to CircleCI 2.0.

jnothman committed:
Don't need no pandas

jnothman commented:
I don't expect changes to 0.19.X after July, so let's not worry about
Circle 2.

I'm also ambivalent about those other examples fixes...

I've made plot_stock_market run without pandas.

jnothman commented:
The remaining issue is fetch_mldata

jnothman commented:
So the thing is, mldata is working fine but is, I think, rejecting requests from Circle.

qinhanmin2014 commented:
Is it possible that migrating to Circle 2 will solve the problem?

qinhanmin2014 commented:
ping @lesteve for insights about Circle CI since it's a blocker.

lesteve commented:
Locally I get the same errors as on the CircleCI build, so maybe something is wrong with mldata:

```py
from sklearn.datasets import fetch_mldata
fetch_mldata('MNIST original')
```

```pytb
---------------------------------------------------------------------------
ConnectionResetError                      Traceback (most recent call last)
<ipython-input-2-f83a0afd5397> in <module>()
----> 1 fetch_mldata('MNIST original')

~/dev/scikit-learn/sklearn/datasets/mldata.py in fetch_mldata(dataname, target_name, data_name, transpose_data, data_home)
    152         urlname = MLDATA_BASE_URL % quote(dataname)
    153         try:
--> 154             mldata_url = urlopen(urlname)
    155         except HTTPError as e:
    156             if e.code == 404:

~/miniconda3/lib/python3.6/urllib/request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    221     else:
    222         opener = _opener
--> 223     return opener.open(url, data, timeout)
    224 
    225 def install_opener(opener):

~/miniconda3/lib/python3.6/urllib/request.py in open(self, fullurl, data, timeout)
    524             req = meth(req)
    525 
--> 526         response = self._open(req, data)
    527 
    528         # post-process response

~/miniconda3/lib/python3.6/urllib/request.py in _open(self, req, data)
    542         protocol = req.type
    543         result = self._call_chain(self.handle_open, protocol, protocol +
--> 544                                   '_open', req)
    545         if result:
    546             return result

~/miniconda3/lib/python3.6/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    502         for handler in handlers:
    503             func = getattr(handler, meth_name)
--> 504             result = func(*args)
    505             if result is not None:
    506                 return result

~/miniconda3/lib/python3.6/urllib/request.py in http_open(self, req)
   1344 
   1345     def http_open(self, req):
-> 1346         return self.do_open(http.client.HTTPConnection, req)
   1347 
   1348     http_request = AbstractHTTPHandler.do_request_

~/miniconda3/lib/python3.6/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1319             except OSError as err: # timeout error
   1320                 raise URLError(err)
-> 1321             r = h.getresponse()
   1322         except:
   1323             h.close()

~/miniconda3/lib/python3.6/http/client.py in getresponse(self)
   1329         try:
   1330             try:
-> 1331                 response.begin()
   1332             except ConnectionError:
   1333                 self.close()

~/miniconda3/lib/python3.6/http/client.py in begin(self)
    295         # read until we get a non-100 response
    296         while True:
--> 297             version, status, reason = self._read_status()
    298             if status != CONTINUE:
    299                 break

~/miniconda3/lib/python3.6/http/client.py in _read_status(self)
    256 
    257     def _read_status(self):
--> 258         line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
    259         if len(line) > _MAXLINE:
    260             raise LineTooLong("status line")

~/miniconda3/lib/python3.6/socket.py in readinto(self, b)
    584         while True:
    585             try:
--> 586                 return self._sock.recv_into(b)
    587             except timeout:
    588                 self._timeout_occurred = True

ConnectionResetError: [Errno 104] Connection reset by peer
```

jnothman commented:
You're quite right. I was mistaken before, when I forgot to set data_home='/tmp'.

I get the same issue.

Maybe we can download the same data from OpenML.

```py
data = scipy.io.arff.loadarff(urllib.request.urlopen('https://www.openml.org/data/download/52667/mnist_784.arff', encoding='utf8'))
data = data.view([('pixels', '<f8', 784), ('class', '|S1')])
y = data['class']
X = data['pixels']
```

jnothman committed:
Replace the use of fetch_mldata in examples

Should also be used in master, although there a cache seems to be preventing runtime failure

sklearn-lgtm commented:
This pull request **fixes 1 alert** when merging 524f6f773c92093f554113d97579a6a254b091df into 7c3944f59338ffd4a49d9aeb658695001dfa1da9 - [view on LGTM.com](https://lgtm.com/projects/g/scikit-learn/scikit-learn/rev/pr-68beb3c0c28180cdafda1bb249fb072cf2693ae0)

**fixed alerts:**

* 1 for Result of integer division may be truncated

---

*Comment posted by [LGTM.com](https://lgtm.com)*

jnothman committed:
I don't get why urlopen is resulting in an AttributeError

sklearn-lgtm commented:
This pull request **fixes 1 alert** when merging 149e54e6ae549741491154656d54bb7e3c95e095 into 7c3944f59338ffd4a49d9aeb658695001dfa1da9 - [view on LGTM.com](https://lgtm.com/projects/g/scikit-learn/scikit-learn/rev/pr-311b23521d00289eef28f043305e0dcc6e21d2ca)

**fixed alerts:**

* 1 for Result of integer division may be truncated

---

*Comment posted by [LGTM.com](https://lgtm.com)*

jnothman committed:
Don't use six...

sklearn-lgtm commented:
This pull request **fixes 1 alert** when merging 965e0c7def5c989aa55d2bfbdbbdde5c9953c381 into 7c3944f59338ffd4a49d9aeb658695001dfa1da9 - [view on LGTM.com](https://lgtm.com/projects/g/scikit-learn/scikit-learn/rev/pr-1ee95c25dbb40671b87146710afdcf355389a4d0)

**fixed alerts:**

* 1 for Result of integer division may be truncated

---

*Comment posted by [LGTM.com](https://lgtm.com)*

jnothman commented:
Yay, it pases!

@ogrisel please feel free to merge and release concurrently. This change shouldn't affect the wheels.

ogrisel commented:
@jnothman thanks, I backported the fix to remove the DeprecationWarning when importing sklearn on Python 3.7. As soon as the wheels are ready I will merge this PR.
